shell/store/index.js

```diff
@@ -404,6 +404,10 @@ export const getters = {
     return state.namespaceFilters;
   },
 
+  allNamespaces(state) {
```

Comment:
I understand why you want this property, but the name is not good. This property exists to create a filtered list of namespaces; it should not be called _all_ Namespaces. Please also add inline comments explaining its function.

Reply:
My only remaining uncertainty with this PR is this getter and associated mutation, specifically because namespaces are already complicated by the cache used for NamespaceFilter - it should be clear to other developers that this is a different form of filtering without having to read through all the code supporting `NamespaceFilter`. Inline comments explaining eg "allNamespaces/changeAllNamespaces allow products to restrict the namespaces shown to the user in the NamespaceFilter and NameNsDescription components."

`allNamespaces` is deceptive because this is not necessarily the same as eg `getters[{"cluster/all"](NAMESPACE)`. 'viewableNamespaces' is more cumbersome but less ambiguous. But I'm more concerned with comments clarifying how/when to use `allNamespaces` and `changeAllNamespaces`